feat: add version outdated metric

qedgardo opened the pull request:
Add new metric solana_node_version_outdated to track if node version is below foundation's minimum required version. The metric returns:
- 1 if node version is outdated
- 0 if node version is current

andreclaro commented on cmd/solana-exporter/collector.go:
```diff
@@ -360,46 +354,124 @@ func (c *SolanaCollector) collectHealth(ctx context.Context, ch chan<- prometheu
 	return
 }
 
+func (c *SolanaCollector) collectVersion(ctx context.Context, ch chan<- prometheus.Metric) {
```
why did you moved this one? to be in order?

qedgardo replied:
to have collectVersion next to collectVersionCOmparision. Should I leave it as it was?

andreclaro commented on cmd/solana-exporter/collector.go:
```diff
@@ -360,46 +354,124 @@ func (c *SolanaCollector) collectHealth(ctx context.Context, ch chan<- prometheu
+	c.logger.Info("Version comparison collected.")
+}
+
+type version struct {
```
move this to the top

andreclaro commented on cmd/solana-exporter/collector.go:
```diff
@@ -360,46 +354,124 @@ func (c *SolanaCollector) collectHealth(ctx context.Context, ch chan<- prometheu
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	c.collectHealth(ctx, ch)
```
why are you changing this? 🤔

qedgardo replied:
this is an error, I shouldn't change that.

andreclaro commented on cmd/solana-exporter/collector.go:
```diff
@@ -360,46 +354,124 @@ func (c *SolanaCollector) collectHealth(ctx context.Context, ch chan<- prometheu
+	c.logger.Info("Version collected.")
+}
+
+func (c *SolanaCollector) collectVersionComparison(ctx context.Context, nodeVersion, minVersion string, ch chan<- prometheus.Metric) {
```
probably the name is not the best... `collectNodeVersionOutdated` ?

Commit:
feat: add version outdated metric

Add new metric solana_node_version_outdated to track if node version is below
foundation's minimum required version. The metric returns:
- 1 if node version is outdated
- 0 if node version is current
